### PR TITLE
Improve UX

### PR DESCRIPTION
**Pull request by flepied**
- split README.md in README.md and CONTRIBUTING.md
- 2 options to consume: use container images from docker hub  or from source code
- add container config into `.env` to make sure nobody needs to edit `docker-compose.yml` files.
- make the README more compelling by explaining the benefits and philosophy and add 2 screenshots.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,11 +59,15 @@ The process below describes how to index changes from a Github repository, a ful
...
 $ mkdir data etc dump
+$ ln -s docker-compose.yml.img docker-compose.yml
 ```
 
 ### Create the config.yaml file
 
-Generate a personal access token on Github (w/o any specific rights).
+The `config.yaml` file is used by the crawler and api services.
+
+If you want to crawl Gihub repositories, generate a personal access
```
Comment (morucci) [inline README.md:69]: Github

File: README.md [hunk 2]
```diff
@@ -58,6 +106,31 @@ tenants:
...
+`MONOCLE_HOST=<host or ip>` into the `.env` file. If you don't
+configure it, `MONOCLE_HOST` is set to `localhost`.
+
+If you want to use a specific version, add `MONOCLE_VERSION=<version>`
+into the `.env` file.
+
+### GitHub authentication
+
+If you want to protect the access to your indices, you can require a
+GitHub login to access and the people able to use the indeices will be
```
Comment (morucci) [inline README.md:121]: Indices

Comment (morucci) [reply README.md:121]: We should define which term we use: indice or index ?

File: docker-compose.yml.img [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  api:
+    image: "changemetrics/monocle_backend:${MONOCLE_VERSION:-latest}"
+    depends_on:
+      - elastic
+    ports:
+      - "9876:9876"
+    environment:
+      CONFIG: /etc/monocle/config.yaml
+      ELASTIC_CONN: elastic:9200
+      ALLOW_ORIGIN: "http://${MONOCLE_HOST:-localhost}:3000"
```
Comment (morucci) [inline docker-compose.yml.img:29]: Can be removed

File: docker-compose.yml.img [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    image: "changemetrics/monocle_backend:${MONOCLE_VERSION:-latest}"
+    depends_on:
+      - elastic
+    command: monocle --elastic-conn elastic:9200 crawler --config /etc/monocle/config.yaml
+    volumes:
+      - $PWD/etc:/etc/monocle:Z
+      - $PWD/dump:/var/lib/crawler:Z
+
+  web:
+    build: web
```
Comment (morucci) [inline docker-compose.yml.img:55]: Should use image instead

File: README.md (hunk 2 above)
Comment (flepied) [reply README.md:121]: in fact indices and indexes are correct plural form of index according to https://grammarist.com/usage/indexes-indices/

File: README.md [hunk 5]
```diff
@@ -1,10 +1,53 @@
...
+The philosophy is to let you visualize and explore metrics that are
+relevant to the way you work.
 
-## Deploy the master version
+For example, your team may want to know:
+
+- what is the ratio of created changes vs merged changes?
+- is there a good balance between change creation and change reviews?
+- what is the ratio of abandoned changes?
+- what are the collaboration patterns between the team members?
```
[the remark this answers is not in the document]
Comment (flepied) [reply README.md:14]: nope :-)

File: README.md [hunk 6]
```diff
@@ -1,10 +1,53 @@
...
 
-![Screenshot](https://user-images.githubusercontent.com/529708/78028243-fc958980-735e-11ea-8fd4-f5ecfb6af02a.png)
+The philosophy is to let you visualize and explore metrics that are
+relevant to the way you work.
 
-## Deploy the master version
+For example, your team may want to know:
+
+- what is the ratio of created changes vs merged changes?
+- is there a good balance between change creation and change reviews?
```
[the remark this answers is not in the document]
Comment (flepied) [reply README.md:12]: ok

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
+# Contributing
+
+## Understanding the design choices
+
+Follow the [Architectural Decision Records](doc/adr/index.md) to
+understand the choices made by the project.
+
+## Use source based docker-compose.yml
+
+To use `docker-compose.yml` from source code, use
```
[the remark this answers is not in the document]
Comment (flepied) [reply CONTRIBUTING.md:10]: ok

File: web/Dockerfile [hunk 8]
```diff
@@ -33,5 +33,20 @@ RUN npm install
...
+
+FROM node:12-alpine
+
+# set working directory
+WORKDIR /app
+
+# install serve globally
+RUN npm install -g serve
+
+COPY --from=0 /app/build /app
```
[the remark this answers is not in the document]
Comment (flepied) [reply web/Dockerfile:49]: It's to do multi-staged build: https://docs.docker.com/develop/develop-images/multistage-build/

File: Dockerfile [hunk 9]
```diff
@@ -17,9 +17,13 @@
...
+RUN apk --no-cache add build-base libffi-dev openssl-dev
 RUN pip install -r requirements.txt
-RUN mkdir /etc/monocle
 COPY monocle monocle
 COPY setup.py setup.py
 RUN python setup.py install
+
+FROM python:3.7-alpine
+RUN mkdir /etc/monocle
+COPY --from=0 /usr/local/lib/python3.7/site-packages /usr/local/lib/python3.7/site-packages
```
[the remark this answers is not in the document]
Comment (flepied) [reply Dockerfile:28]: to remove build dependencies.

File: README.md [hunk 10]
```diff
@@ -1,10 +1,53 @@
...
+- what is the ratio of abandoned changes?
+- what are the collaboration patterns between the team members?
+- how long does it take to merge a change?
+- average delay for the first comment or review?
+- long standing changes?
+- do we have new contributors?
+
+The main idea behind Monocle is to detect anomalies in your development process.
```
[the remark this answers is not in the document]
Comment (flepied) [reply README.md:20]: ok

File: README.md (hunk 1 above)
Comment (flepied) [reply README.md:69]: ok

File: docker-compose.yml.img (hunk 3 above)
Comment (flepied) [reply docker-compose.yml.img:29]: I don't get it. Why?

File: docker-compose.yml.img (hunk 4 above)
Comment (flepied) [reply docker-compose.yml.img:55]: ok

**Pull request merged**